virttest/qemu_vm.py

```diff
@@ -1107,7 +1107,25 @@ def add_qxl(qxl_nr, base_addr=29):
             return qxl_str
 
         def add_vga(vga):
-            return " -vga %s" % vga
+            """Add primary vga device."""
+            pci_host_bus = {'aobject': 'pci.0'}
+            vga_dev_map = {
+                # name: (driver, parent bus)
+                "std": ("VGA", pci_host_bus),
+                "cirrus": ("cirrus-vga", pci_host_bus),
+                "vmware": ("vmware-svga", pci_host_bus),
+                "qxl": ("qxl-vga", pci_host_bus),
+                "virtio": ("virtio-vga", pci_host_bus),
+            }
+            vga_dev, parent_bus = vga_dev_map.get(vga, (None, None))
```

Comment:
Hmm, `-vga None`... that will not work well. I'd simply raise an exception saying `"unsupported '%s' vga type, please update mapping" % vga` or even better simply use `vga` as value for the device.

Reply:
Actually, if `vga_dev` gets `None` here, it will fallback to the legacy expression as line 1122 has already done the check, and there is another check in line 1485 to prevent getting `vga=None` in this function, so user won't see `-vga None` in command line.

Reply:
Oh I see. I'd still prefer to be able to specify custom device (imagine a developer working on a new type) instead of automatic fallback. Anyway it's just a mild suggestion, not a strong request.

Reply:
sure, but we can assume default as PCI and for `mmio` based machines no bus at all... (and only think on perfect solution when this proves to be wrong)

Reply:
Please note that we also have ISA vga devices (and not sure whether there would be other type of devices in the future), so I don't really want a complicated automatic bus assigning logic since we already have a mapping table to do this in a more accurate way.